test/Driver/linker.swift

```diff
@@ -90,21 +92,37 @@
 // watchOS_SIMPLE: -o linker
 
 
-// LINUX: swift
-// LINUX: -o [[OBJECTFILE:.*]]
-
-// LINUX: clang++{{"? }}
-// LINUX-DAG: [[OBJECTFILE]]
-// LINUX-DAG: -lswiftCore
-// LINUX-DAG: -L [[STDLIB_PATH:[^ ]+/lib/swift]]
-// LINUX-DAG: -Xlinker -rpath -Xlinker [[STDLIB_PATH]]
-// LINUX-DAG: -Xlinker -T /{{[^ ]+}}/linux/x86_64/swift.ld
-// LINUX-DAG: -F foo
-// LINUX-DAG: -framework bar
-// LINUX-DAG: -L baz
-// LINUX-DAG: -lboo
-// LINUX-DAG: -Xlinker -undefined
-// LINUX: -o linker
+// LINUX-x86_64: swift
+// LINUX-x86_64: -o [[OBJECTFILE:.*]]
+
+// LINUX-x86_64: clang++{{"? }}
+// LINUX-x86_64-DAG: [[OBJECTFILE]]
+// LINUX-x86_64-DAG: -lswiftCore
+// LINUX-x86_64-DAG: -L [[STDLIB_PATH:[^ ]+/lib/swift]]
+// LINUX-x86_64-DAG: -Xlinker -rpath -Xlinker [[STDLIB_PATH]]
+// LINUX-x86_64-DAG: -Xlinker -T /{{[^ ]+}}/linux/x86_64/swift.ld
+// LINUX-x86_64-DAG: -F foo
+// LINUX-x86_64-DAG: -framework bar
+// LINUX-x86_64-DAG: -L baz
+// LINUX-x86_64-DAG: -lboo
+// LINUX-x86_64-DAG: -Xlinker -undefined
+// LINUX-x86_64: -o linker
+
+// LINUX-armv7: swift
+// LINUX-armv7: -o [[OBJECTFILE:.*]]
+
+// LINUX-armv7: clang++{{"? }}
+// LINUX-armv7-DAG: [[OBJECTFILE]]
+// LINUX-armv7-DAG: -lswiftCore
+// LINUX-armv7-DAG: -L [[STDLIB_PATH:[^ ]+/lib/swift]]
+// LINUX-armv7-DAG: -Xlinker -rpath -Xlinker [[STDLIB_PATH]]
+// LINUX-armv7-DAG: -Xlinker -T /{{[^ ]+}}/linux/arm/swift.ld
```

Comment:
Ah, no, it does not pass because the whole file is `// XFAIL: linux`.

Feel free to factor out Linux-specific or Apple-specific parts into a separate file to increase test coverage.